File: HolyDay/Models/Emotion.swift
//
//  Emotion.swift
//  HolyDay
//
//  Created by Matthias Cadet on 31/05/2026.
//

import SwiftUI

/// État intérieur que l'utilisateur déclare avant de prier.
/// La `rawValue` est stable et non localisée : c'est elle qui est persistée sur `PrayerEntry`.
enum Emotion: String, CaseIterable, Codable, Identifiable {
  case joy
  case peace
  case gratitude
  case sadness
  case fear
  case fatigue
  case anger
  case hope

  var id: String { rawValue }

  var titleKey: LocalizedStringKey {
    switch self {
    case .joy: "emotion.joy"
    case .peace: "emotion.peace"
    case .gratitude: "emotion.gratitude"
    case .sadness: "emotion.sadness"
    case .fear: "emotion.fear"
    case .fatigue: "emotion.fatigue"
    case .anger: "emotion.anger"
    case .hope: "emotion.hope"
    }
  }

  var accessibilityLabel: String {
    switch self {
    case .joy: String(localized: "emotion.joy")
    case .peace: String(localized: "emotion.peace")
    case .gratitude: String(localized: "emotion.gratitude")
    case .sadness: String(localized: "emotion.sadness")
    case .fear: String(localized: "emotion.fear")
    case .fatigue: String(localized: "emotion.fatigue")
    case .anger: String(localized: "emotion.anger")
    case .hope: String(localized: "emotion.hope")
    }
  }

  var icon: String {
    switch self {
    case .joy: "sun.max.fill"
    case .peace: "leaf.fill"
    case .gratitude: "hands.and.sparkles.fill"
    case .sadness: "cloud.rain.fill"
    case .fear: "wind"
    case .fatigue: "moon.zzz.fill"
    case .anger: "flame.fill"
    case .hope: "sunrise.fill"
    }
  }

  /// Nom de couleur résolu par `AppTheme.color(for:)`, aligné sur la palette ACTS.
  var colorName: String {
    switch self {
    case .joy, .gratitude: "thanksgivingGold"
    case .peace, .sadness: "confessionBlue"
    case .fear: "adorationPurple"
    case .fatigue, .hope: "supplicationGreen"
    case .anger: "adaptiveOrange"
    }
  }

  var color: Color { AppTheme.color(for: colorName) }
}

File: android/app/src/main/java/com/matthiascadet/holyday/data/model/Emotion.kt
package com.matthiascadet.holyday.data.model

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Air
import androidx.compose.material.icons.filled.Bedtime
import androidx.compose.material.icons.filled.LocalFireDepartment
import androidx.compose.material.icons.filled.Spa
import androidx.compose.material.icons.filled.Umbrella
import androidx.compose.material.icons.filled.VolunteerActivism
import androidx.compose.material.icons.filled.WbSunny
import androidx.compose.material.icons.filled.WbTwilight
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import com.matthiascadet.holyday.R
import com.matthiascadet.holyday.ui.theme.AppTheme
import com.matthiascadet.holyday.ui.theme.BrandColors

/**
 * État intérieur que l'utilisateur déclare avant de prier.
 * `id` est stable et non localisé : c'est lui qui est persisté sur `PrayerEntry`
 * (équivalent de la `rawValue` de l'enum Swift `Emotion`).
 */
enum class Emotion(val id: String, val titleRes: Int, val icon: ImageVector, val colorName: String, val pastel: Color) {
    JOY("joy", R.string.emotion_joy, Icons.Filled.WbSunny, "thanksgivingGold", BrandColors.emotionJoy),
    PEACE("peace", R.string.emotion_peace, Icons.Filled.Spa, "confessionBlue", BrandColors.emotionPeace),
    GRATITUDE("gratitude", R.string.emotion_gratitude, Icons.Filled.VolunteerActivism, "thanksgivingGold", BrandColors.emotionGratitude),
    SADNESS("sadness", R.string.emotion_sadness, Icons.Filled.Umbrella, "confessionBlue", BrandColors.emotionSadness),
    FEAR("fear", R.string.emotion_fear, Icons.Filled.Air, "adorationPurple", BrandColors.emotionFear),
    FATIGUE("fatigue", R.string.emotion_fatigue, Icons.Filled.Bedtime, "supplicationGreen", BrandColors.emotionFatigue),
    ANGER("anger", R.string.emotion_anger, Icons.Filled.LocalFireDepartment, "adaptiveOrange", BrandColors.emotionAnger),
    HOPE("hope", R.string.emotion_hope, Icons.Filled.WbTwilight, "supplicationGreen", BrandColors.emotionHope);

    @Composable
    fun color(): Color = AppTheme.colorFor(colorName)

    companion object {
        fun fromId(id: String?): Emotion? = id?.let { value -> entries.find { it.id == value } }
    }
}
